src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java

```diff
@@ -347,6 +350,15 @@ public long retryInterval() {
         return retryInterval;
     }
 
+    public C sessionProvider(SessionProvider sessionProvider) {
+        this.sessionProvider = sessionProvider;
+        return castThis();
+    }
+
+    public SessionProvider sessionProvider() {
```

Comment:
Should we add some JavaDoc here? it feels like the kind of thing that if not understood could cause some head scratching. e.g. "when configured the session details will be the first thing sent on connection establishment"